__tests__/TemplateIndexer.test.js

```diff
@@ -49,4 +49,127 @@ describe('Indexer', () => {
+            {
+              '@id': 'sinopia:resourceTemplate:schema:Thing'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasClass': [
+            {
+              '@id': 'http://schema.org/Thing'
+            }
+          ],
+          'http://www.w3.org/2000/01/rdf-schema#label': [
+            {
+              '@value': 'The most generic type of item'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasAuthor': [
+            {
+              '@value': 'Shelia Authority'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyTemplate': [
+            {
+              '@list': [
+                {
+                  '@id': '_:b12415'
+                },
+                {
+                  '@id': '_:b12417'
+                }
+              ]
+            }
+          ]
+        },
+        {
+          '@id': '_:b12415',
+          '@type': [
+            'http://sinopia.io/vocabulary/PropertyTemplate'
+          ],
+          'http://www.w3.org/2000/01/rdf-schema#label': [
+            {
+              '@value': 'Additional Type'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyUri': [
+            {
+              '@id': 'http://schema.org/additionalType'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyAttribute': [
+            {
+              '@id': 'http://sinopia.io/vocabulary/propertyAttribute/repeatable'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyType': [
+            {
+              '@id': 'http://sinopia.io/vocabulary/propertyType/literal'
+            }
+          ]
+        },
+        {
+          '@id': '_:b12417',
+          '@type': [
+            'http://sinopia.io/vocabulary/PropertyTemplate'
+          ],
+          'http://www.w3.org/2000/01/rdf-schema#label': [
+            {
+              '@value': 'Alternative Name'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyUri': [
+            {
+              '@id': 'http://schema.org/alternateName'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyAttribute': [
+            {
+              '@id': 'http://sinopia.io/vocabulary/propertyAttribute/repeatable'
+            }
+          ],
+          'http://sinopia.io/vocabulary/hasPropertyType': [
+            {
+              '@id': 'http://sinopia.io/vocabulary/propertyType/literal'
+            }
+          ]
+        }
+      ]
+    }
+    it('indexes a resource template', async () => {
+      const dataset = await datasetFromJsonld(doc.data)
+      const templateIndexer = new TemplateIndexer(doc, dataset)
+      const indexObject = templateIndexer.index()
+      expect(indexObject.id).toBe('sinopia:resourceTemplate:schema:Thing')
+      expect(indexObject.author).toBe('Shelia Authority')
+      expect(indexObject.date).toBeUndefined()
```

Comment:
Can you say why the date would be undefined? I guess it's not derived from the timestamp k/v pair?

Reply:
> Can you say why the date would be undefined? I guess it's not derived from the timestamp k/v pair?

The RDF payload does not have a triple with a <http://sinopia.io/vocabulary/hasDate> predicate (see https://github.com/LD4P/sinopia_indexing_pipeline/blob/6ede7b5c70fd09d2c70581063b1422eacb1da040/src/TemplateIndexer.js#L27). Happy to add this triple to the test fixture but I wanted to exercise the conditional in the `valueFor` function that returns undefined if not found.

Reply:
Sounds good, thanks!